fdbclient/FileBackupAgent.actor.cpp

```diff
@@ -4069,10 +4073,29 @@ struct StartFullRestoreTaskFunc : RestoreTaskFuncBase {
 		// Order does not matter, they will be put in order when written to the restoreFileMap below.
 		state std::vector<RestoreConfig::RestoreFile> files;
 
+		state Version firstConsistentVersion = -1;
```

Comment:
I see a problem here that we missed on the master branch, it will have to be fixed there first.

If `inconsistentSnapshotOnly` is set then there probably is no first consistent version as the restore will probably never be consistent.  The exception to this is if there is exactly 1 range file then technically the restore is consistent as of its version once it is loaded.  If there are more than 1 range files, then they would all have to be at the same version which is extremely unlikely.

Reply:
I think it is reasonable to fix this by leaving `firstConsistentVersion` at -1 if `inconsistentSnapshotOnly` is set unless there is only 1 range file, in which case it can be set to that file's version.

Reply:
This can be on a separate PR on master, then once merged its commit can cherry-picked to this PR.

Reply:
Since this can be on a separate PR, I'll dismiss this change request so this PR can be merged.